src/client/Microsoft.Identity.Client.Brokers/WamAdapters.cs

```diff
@@ -0,0 +1,236 @@
+            int errorCode;
+
+            switch ((int)authResult.Error.Status)
+            {
+                case (int)ResponseStatus.UserCanceled:
+                    logger.Verbose($"[WamBroker] {MsalError.AuthenticationCanceledError} {MsalErrorMessage.AuthenticationCanceled}");
+                    throw new MsalClientException(MsalError.AuthenticationCanceledError, MsalErrorMessage.AuthenticationCanceled);
+
+                case (int)ResponseStatus.InteractionRequired:
+                case (int)ResponseStatus.AccountUnusable:
+                    errorCode = authResult.Error.ErrorCode;
+                    internalErrorCode = authResult.Error.Tag.ToString(CultureInfo.InvariantCulture);
+                    errorMessage = 
+                        $"{WamErrorPrefix} \n" +
+                        $" Error Code: {errorCode} \n" +
+                        $" Error Message: {authResult.Error.Context} \n" +
+                        $" Internal Error Code: {internalErrorCode} \n";
+                    logger.Verbose($"[WamBroker] {MsalError.FailedToAcquireTokenSilentlyFromBroker} {errorMessage}");
+                    throw new MsalUiRequiredException(MsalError.FailedToAcquireTokenSilentlyFromBroker, errorMessage);
+
+                case (int)ResponseStatus.IncorrectConfiguration:
+                case (int)ResponseStatus.ApiContractViolation:
+                    errorCode = authResult.Error.ErrorCode;
+                    internalErrorCode = (authResult.Error.Tag).ToString(CultureInfo.InvariantCulture);
+                    errorMessage =
+                        $"{WamErrorPrefix} \n" +
+                        $" Error Code: {errorCode} \n" +
+                        $" Error Message: {authResult.Error.Status} \n" +
+                        $" WAM Error Message: {authResult.Error.Context} \n" +
+                        $" Internal Error Code: {internalErrorCode} \n" +
+                        $" Is Retryable: false \n" +
+                        $" Possible causes: \n" +
+                        $"- Invalid redirect uri - ensure you have configured the following url in the AAD portal App Registration: {WamAdapters.GetExpectedRedirectUri(authenticationRequestParameters.AppConfig.ClientId)} \n" +
+                        $"- No Internet connection \n" +
+                        $"Please see https://aka.ms/msal-net-wam for details about Windows Broker integration";
+                    logger.Verbose($"[WamBroker] WAM_provider_error_{errorCode} {errorMessage}");
+                    serviceException = new MsalServiceException($"WAM_provider_error_{errorCode}", errorMessage);
+                    serviceException.IsRetryable = false;
+                    throw serviceException;
+
+                case (int)ResponseStatus.NetworkTemporarilyUnavailable:
+                case (int)ResponseStatus.NoNetwork:
+                case (int)ResponseStatus.ServerTemporarilyUnavailable:
+                    errorCode = authResult.Error.ErrorCode;
```

Comment:
looks like `errorCode` and `internalErrorCode` are always set to the same values in every case. Consider moving them outside the `switch`

Reply:
moved